numpyro/distributions/directional.py

```diff
@@ -0,0 +1,62 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+import math
+
+import jax.numpy as jnp
+from jax import lax, dtypes
+
+from numpyro.distributions import constraints
+from numpyro.distributions.distribution import Distribution
+from numpyro.distributions.util import validate_sample
+from numpyro.util import copy_docs_from
+from numpyro.distributions.util import promote_shapes, von_mises_centered
+
+
+@copy_docs_from(Distribution)
+class VonMises(Distribution):
+    arg_constraints = {'loc': constraints.interval(-math.pi, math.pi), 'concentration': constraints.positive}
+
+    @property
+    def support(self):
+        return constraints.interval(-math.pi, math.pi)
+
+    def __init__(self, loc, concentration, validate_args=None):
+        self.loc, self.concentration = promote_shapes(loc, concentration)
+
+        batch_shape = lax.broadcast_shapes(jnp.shape(concentration), jnp.shape(loc))
+
+        super(VonMises, self).__init__(batch_shape=batch_shape,
+                                       validate_args=validate_args)
+
+    def sample(self, key, sample_shape=()):
+        """ Generate sample from von Mises distribution
+
+            :param sample_shape: shape of samples
+            :param rng_key: random number generator key
+            :return: samples from von Mises
+        """
+        samples = von_mises_centered(key, self.concentration, sample_shape, dtypes.canonicalize_dtype(jnp.float64))
```

Comment:
I think you need to use `shape=sample_shape + self.shape()` here. To detect those issues, you can add more test cases instead of just the scalar one `T(dist.VonMises, 2., 10.)`.

nit: you can omit `canonicalize_dtype` stuff here because it should be handled in `von_mises_centered`

Reply:
I'll add some more tests, and update as needed 👍